### PR TITLE
BaSP-TG-43: super admin entity

### PR DESCRIPTION
**Pull request by Ansitox**

### CODE REVIEW

File: src/Components/SuperAdmins/Form/Form.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      const id = url.substring(url.lastIndexOf('=') + 1);
+      let response = await fetch(`${process.env.REACT_APP_API_URL}/superAdmin/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(input)
+      });
+      console.log(response);
+      if (response.status === 200) {
```
Comment (matescarabino) [inline src/Components/SuperAdmins/Form/Form.js:87]: wouldn't it be 201?

**Comment by matescarabino** [comment]
It looks fine to me

File: src/Components/SuperAdmins/Form/Form.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      const url = window.location.href;
+      const id = url.substring(url.lastIndexOf('=') + 1);
+      let response = await fetch(`${process.env.REACT_APP_API_URL}/superAdmin/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(input)
+      });
+      console.log(response);
```
Comment (matescarabino) [inline src/Components/SuperAdmins/Form/Form.js:86]: Remove before merge

File: src/Components/SuperAdmins/index.js [hunk 3]
```diff
@@ -1,11 +1,58 @@
...
+const SuperAdminsList = () => {
+  const formURL = './super-admins/form';
+  const [superAdminsList, setSuperAdminsList] = useState([]);
+  const getList = async () => {
+    try {
+      let response = await fetch(`${process.env.REACT_APP_API_URL}/superAdmin/`);
+      response = await response.json();
+      setSuperAdminsList(response.data);
+    } catch (error) {
+      console.log(error);
```
Comment (matescarabino) [inline src/Components/SuperAdmins/index.js:14]:
```suggestion
      console.error(error);
```

File: src/Components/SuperAdmins/Form/Form.js (hunk 2 above)
Comment (Ansitox) [reply src/Components/SuperAdmins/Form/Form.js]: Done.

File: src/Components/SuperAdmins/Form/Form.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    }
+  };
+
+  useEffect(async () => {
+    if (window.location.href.includes('id=')) {
+      currentSuperAdminInput();
+    }
+  }, []);
+
+  const handler = () => {
```
Comment (ignacioaristo) [inline src/Components/SuperAdmins/Form/Form.js:39]: we need be more specific for this function's name

File: src/Components/SuperAdmins/List/List.module.css [hunk 5]
```diff
@@ -0,0 +1,62 @@
+th {
+    text-align: start;
+}
+
+.table {
```
Comment (ignacioaristo) [inline src/Components/SuperAdmins/List/List.module.css:5]: move all the tag selector above. Then the classes.

File: src/Components/SuperAdmins/index.js [hunk 6]
```diff
@@ -1,11 +1,57 @@
+import { useEffect, useState } from 'react';
+import List from './List/List';
 import styles from './super-admins.module.css';
 
-function SuperAdmins() {
+const SuperAdminsList = () => {
+  const [superAdminsList, setSuperAdminsList] = useState([]);
+  const [err, setError] = useState('');
```
Comment (ignacioaristo) [inline src/Components/SuperAdmins/index.js:7]: In order to follow good practices, name the setState function `setErr`

File: src/Components/SuperAdmins/index.js [hunk 7]
```diff
@@ -1,11 +1,57 @@
...
+      let response = await fetch(`${process.env.REACT_APP_API_URL}/superAdmin/`);
+      response = await response.json();
+      setSuperAdminsList(response.data);
+    } catch (error) {
+      setError(error);
+      alert(err);
+    }
+  };
+
+  useEffect(async () => {
```
Comment (ignacioaristo) [inline src/Components/SuperAdmins/index.js:20]: You are handle the catch here, but you are handling the error when you created, so, in this useEffect you only have to invoke the function.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]
Change all the components to arrow functions

File: src/Components/SuperAdmins/index.js (hunk 7 above)
Comment (Ansitox) [reply src/Components/SuperAdmins/index.js:20]: Done.

File: src/Components/SuperAdmins/index.js (hunk 6 above)
Comment (Ansitox) [reply src/Components/SuperAdmins/index.js]: Done.

File: src/Components/SuperAdmins/List/List.module.css (hunk 5 above)
Comment (Ansitox) [reply src/Components/SuperAdmins/List/List.module.css]: Done.

File: src/Components/SuperAdmins/Form/Form.js (hunk 4 above)
Comment (Ansitox) [reply src/Components/SuperAdmins/Form/Form.js]: Done.

**Review by matescarabino — APPROVED** [review]

**Review by MaxiG-dev — APPROVED** [review]
Everything looks great!

**Review by ignacioaristo — APPROVED** [review]

**Review by NicoSan02 — APPROVED** [review]
🕵️‍♂️ **QA FEEDBACK**

**SUPER ADMIN**

First of all, we wanted to let you know that we consider that **your PR complies
the requirements to be approved**, and that you can merge to the master branch 👍

however we consider leaving you some comments:

*when creating or editting, Name, Last Name or Password are invalid, the modal displayed contains a confusing message.

modal error

**Pull request merged**